File: company-service/src/main/java/ru/panic/companyservice/template/repository/CompanyRepository.java
package ru.panic.companyservice.template.repository;

import ru.panic.companyservice.template.entity.Company;

import java.util.List;

public interface CompanyRepository {
    void save(Company company);
    void delete(Company company);
    List<Company> findAllCompanyByOwner(String owner);
    boolean extendedByCompanyName(String companyName);
}

File: company-service/src/main/kotlin/ru/panic/companyservice/template/repository/CompanyRepository.kt
package ru.panic.companyservice.template.repository

import ru.panic.companyservice.template.entity.Company

interface CompanyRepository {
    fun save(company: Company)
    fun findCompanyByCompanyName(companyName: String): Company?
    fun deleteByCompanyName(companyName: String)
    fun extendedBySecretKey(secretKey: String): Boolean
    fun findAllCompanyByOwner(owner: String): List<Company>?
}